bifromq-dist/bifromq-dist-worker/src/main/java/com/baidu/bifromq/dist/worker/SubscriptionCache.java

```diff
@@ -217,6 +218,7 @@ public void invalidate(ScopedTopic topic) {
         if (routeCache != null) {
             routeCache.synchronous().invalidate(topic);
         }
+        orderedSharedMatching.invalidate(topic.tenantId + escape(topic.topic) + tenantVerCache.get(topic.tenantId));
```

Comment:
using a dedicated Cache Key type should provide better performance and encapsulation